src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.ui.Ui;
+
+/**
+ * Reads the users command.
+ */
+public class Parser {
+
+    /**
+     * Reads the user's command, processes it, and returns the appropriate response.
+     *
+     * @param command   The user's input command.
+     * @param tasklist  The task list to which the command should be applied.
+     * @return          The response or result of processing the user's command.
+     */
+    public static String readTask(String command, TaskList tasklist) {
+        String[] commandSplit = command.split(" ");
+        if (command.equals("bye")) {
+            System.exit(0);
+            return Ui.exit();
+        } else if (command.equals("list")) {
+            return Ui.listTasks(tasklist);
+        } else if (command.startsWith("find")) {
+            return tasklist.findTasks(commandSplit[1]);
+        } else {
+            if (command.startsWith("todo") || command.startsWith("deadline")
+                    || command.startsWith("event")) {
+                return addTask(command, tasklist);
+            } else if (command.startsWith("mark") || command.startsWith("unmark")
+                    || command.startsWith("delete")) {
+                try {
+                    if (commandSplit.length > 2 || Integer.parseInt(commandSplit[1]) > tasklist.taskCount()) {
+                        return Ui.showError(new DukeException("Please enter a valid number"));
+                    }
+                    int index = Integer.parseInt(commandSplit[1]) - 1;
+                    if (command.startsWith("mark")) {
+                        return tasklist.markTask(index);
+                    } else if (command.startsWith("unmark")) {
+                        return tasklist.unMarkTask(index);
+                    } else {
+                        return tasklist.deleteTask(index);
+                    }
+                } catch (NumberFormatException e) {
+                    return Ui.showError(new DukeException("Please enter a valid number"));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    return Ui.showError(new DukeException("Please enter a valid number"));
+                }
+            } else {
+                return Ui.showError(new DukeException("I'm sorry, but I don't know what that means :-("));
+            }
+        }
+    }
```

Comment:
Try to avoid deep nesting as it makes the code harder to read. Maybe can consider doing the boolean checks in their own separate methods? Or can consider creating a Command package with all the individual commands?